data/core/common.lua

```diff
@@ -322,4 +322,46 @@ function common.mkdirp(path)
   return true
 end
 
+function common.rmdir(path, recursively)
+  local stat = system.get_file_info(path)
+  if not stat or stat.type ~= "dir" then
+    return false, "given path is not a directory", path
+  end
+
+  local contents = system.list_dir(path)
+  if #contents > 0 and not recursively then
+    return false, "directory is not empty", path
+  end
+
+  if #contents > 0 then
+    for _, item in pairs(contents) do
+      local item_path = path .. PATHSEP .. item
+      local item_stat = system.get_file_info(item_path)
+
+      if not item_stat then
+        return false, "invalid file encountered", item_path
+      end
+
+      if item_stat.type == "dir" then
+        local deleted, error, ipath = common.rmdir(item_path, recursively)
+        if not deleted then
+          return false, error, ipath
+        end
+      elseif item_stat.type == "file" then
+        local removed, error = system.rmdir(item_path)
```

Comment:
We are using system.rmdir to delete a file ?

EDIT: ok, looks like system.rmdir may also remove a file but on Windows ? The names of the functions are so confusing.

Reply:
That was actually a mistake on me, replaced that with os.remove which is the correct one for this.